src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1176,6 +1240,21 @@ private void ImportThrow()
 
         private void ImportLoadField(int token, bool isStatic)
         {
+            if (isStatic)
```

Comment:
The `isStatic` parameter will be true only for `ldsfld`; depending on how production quality code you want to write, this should also check that the field is not static (ECMA-335 allows `ldfld` (`isStatic == false` but the token references a static field), in which case the `this` is ignored).

Reply:
I'd think "production quality" for this code would be implementing static fields instead of throwing. That'll be worth keeping in mind when we go to implement it though. Added a comment to #4532

Reply:
@hippiehunter did take care of it in #4729